newrelic/resource_newrelic_alert_muting_rule.go

```diff
@@ -183,6 +183,12 @@ func resourceNewRelicAlertMutingRule() *schema.Resource {
 				Elem:        scheduleSchema(),
 				Description: "The time window when the MutingRule should actively mute incidents.",
 			},
+			"end_behaviour": {
```

Comment:
@Aashirwadjain 

This is a nice, concise attribute name, but by convention, we need make sure that the name of an attribute in a Terraform resource fully corresponds to what's present in the API, so the customer doesn't find it difficult to understand what `end_behaviour` refers to, while what they are looking for actually is `actionOnMutingRuleWindowEnded`. 

Please rename the attribute accordingly. Ideally, it could be as is in the API though long, `action_on_muting_rule_window_ended`. We have had to add such lengthy attributes at times for API parity in the past too, so this should be okay.

In addition, in cases like these, since you'll now have to rename all references of this attribute, in future (whenever applicable for small-scale changes like these), I would suggest defining attribute names as constants (if there are not too many), similar to this

https://github.com/newrelic/terraform-provider-newrelic/blob/25894e484177876891d4407de299bd4692f20e6e/newrelic/resource_helpers_synthetics.go#L113-L115


and using references to this constant instead of directly writing the string with the attribute name, so it would help in such cases where we'd need to rename attributes. You can add it directly to this resource file, the above is just an example.

Reply:
Thanks for reference @pranav-new-relic ,`end_behaviour` is modified as `action_on_muting_rule_window_ended`